howto/stream/exchange-oob-data.md

```diff
@@ -269,6 +270,50 @@ try {
     ex.printStackTrace();
 }
 ```
+
+#### Install the APK as system app
+
+To connect the data channel to the Anbox WebRTC data proxy service within an Android container, the Android app must be installed and running as a system app. To do so, proceed with the following steps:
+1. Add the attribute `android:sharedUserId="android.uid.system"` into the `<manifest>` tag in the AndroidManifest.xml file of your Android app
+2. Create an addon to install your APK as a system app through the pre-start hook with the following content:
+    ```
+    #!/bin/bash -ex
+
+    # Only install the APK as a system app when bootstrapping an application.
+    if  [ "$INSTANCE_TYPE" = "regular" ]; then
+      exit 0
+    fi
+
+    aam install-system-app \
+      --apk="${ADDON_DIR}"/app.apk \
+      --permissions=<comma-separated list of permissions that the application requires> \
+      --package-name=<package_name>
+      --access-hidden-api
+    ```
+
+   See [How to install an APK as a system app](https://documentation.ubuntu.com/anbox-cloud/en/latest/howto/port/install-apk-system-app/#howto-install-apk-system-app) for details.
+
+3. Navigate to the addon directory and add it to AMS:
+    ```
+    amc addon add install-out-of-band-app .
+    ```
+
+#### Run end-to-end test
+
+Launch an instance with streaming enabled and the addon created above
+```
+amc launch --raw jammy:android12:amd64 \
+  --enable-streaming \
+  --features allow_custom_system_signatures \
+  --addons install-out-of-band-app
+```
+
+```{note}
+Enabling the `allow_custom_system_signatures` feature is required to run the Android application as a system app in an Android container
+```
+
+Once the instance is running, test whether the Android application can connect to the data channel created by the WebRTC client and verify that messages can be successfully sent and received between both ends.
```

Comment:
As a user, how do I test this? It would be good to add or link to existing information.

Reply:
I agree but still listing a couple of steps to run to successfully test this can be useful for someone trying this out for the first time.

Reply:
I just reworded that part.
Please take another look.
Thanks